### DISCUSSION TITLE
Achieve multi-level drawer behavior without actually nesting them.

### DISCUSSION THREAD

**Discussion by yash-sharma01** [ant-design/ant-design#44185] [Q&A]
I'm working on a SaaS product in which we are using multiple drawers throughout the app. The multi-level drawer from Ant Design is superb, but as our codebase scales, it's becoming difficult to maintain the drawers. We need to open the same drawer from multiple positions, so to simplify the code, I've created a context and am handling the drawers using a global state.

This approach solved the maintainability issue, but as we moved drawers to separate files, we lost the multi-level behavior of Ant Design.

I just need help in achieving multi-level behavior without actually nesting them (perhaps with some props or something).

Any help would be greatly appreciated 🙏

**Comment by thevinodpatidar** [comment]
I'm also facing the same issue.
